### PR TITLE
[py-tx] New Names for SignalOpinionCategory

### PR DESCRIPTION
**Pull request by Dcallies**
Summary
---------

Maybe the outcome of the post titled "Open Source Naming Conventions for False Positives and Others". I don't like any of the names, and we are going to see these a whole lot (i.e. output of match command). 

It's important that the names make sense or at least aren't nonsensical

Test Plan
---------

Pending

### CODE REVIEW

**Comment by Dcallies** [comment]
I was inspired by this while working on this: https://developers.google.com/machine-learning/glossary#multi-class-classification

**Comment by Dcallies** [comment]
After thinking about this more, I think what I actually want to do is hide the names from the match output by default - the difference between them won't matter as much for most users, which reduces the pressure to get it perfect.

File: python-threatexchange/threatexchange/exchanges/fetch_state.py [hunk 1]
```diff
@@ -74,15 +74,16 @@ class SignalOpinionCategory(IntEnum):
...
-    WORTH_INVESTIGATING = 1  # Indirect indicator
-    # TODO: Rename to FITS_CATEGORY
-    TRUE_POSITIVE = 2  # Confirmed meets category
+    # Negative class
+    # Signal generates false positives, or is test/example data
+    DOES_NOT_FIT_CATEGORY = 0
+    # Indirect indicator, requires verifying
+    MAY_FIT_CATEGORY = 1
+    # Positive class
+    FITS_CATEGORY = 2
```
Comment (Dcallies) [inline python-threatexchange/threatexchange/exchanges/fetch_state.py:86]: After sleeping on it, I propose:
```suggestion
    NEGATIVE_CLASS = 0
    # Indirect indicator, requires verifying
    INVESTIGATION_SEED = 1
    # Positive class
    POSITIVE_CLASS= 2
    # Maybe in the future: Informational 
```

File: python-threatexchange/threatexchange/cli/match_cmd.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ class MatchCommand(command_base.Command):
     * TRUE_POSITIVE:
       All contributors of this signal believe that matching content should fit
       the collaboration
-    * WORTH_INVESTIGATING:
+    * INVESTIGATION_SEED:
       This content needs manual investigation to confirm or fanout to find the
       content that fits the collaboration
     * DISPUTED:
```
Comment (Dcallies) [inline python-threatexchange/threatexchange/cli/match_cmd.py:67]: Missed

**Review by BarrettOlson — APPROVED** [review]
LGTM, two ignorable comment nits

File: python-threatexchange/threatexchange/cli/match_cmd.py [hunk 3]
```diff
@@ -58,20 +58,18 @@ class MatchCommand(command_base.Command):
...
       Some members have said that this signal can be used to find content
       that fits the collaboration, but others have said it matches content
       that does not belong in the collaboration.
-    * FALSE_POSITIVE:
+    * NEGATIVE_CLASS:
       Members have said content that matches does not belong in the
-      collaboration, or that this is information content that should not
-      trigger normal side-effects (i.e. human review).
+      collaboration, or that this is informational content that should be treated.
```
[the remark this answers is not in the document]
Comment (BarrettOlson) [reply python-threatexchange/threatexchange/cli/match_cmd.py:72]: or ~~that should be treated~~

File: python-threatexchange/threatexchange/exchanges/fetch_state.py [hunk 4]
```diff
@@ -74,15 +74,15 @@ class SignalOpinionCategory(IntEnum):
...
 
-    # TODO: Rename to DOES_NOT_FIT_CATEORY
-    FALSE_POSITIVE = 0  # Signal generates false positives
-    # TODO: Rename to something else
-    WORTH_INVESTIGATING = 1  # Indirect indicator
-    # TODO: Rename to FITS_CATEGORY
-    TRUE_POSITIVE = 2  # Confirmed meets category
+    # Signal generates false positives, or is test/example data
+    NEGATIVE_CLASS = 0
+    # Indirect indicator, requires verifying. Can lead to POSITIVE_CLASS content
```
[the remark this answers is not in the document]
Comment (Dcallies) [reply python-threatexchange/threatexchange/exchanges/fetch_state.py:82]: In this case, I did want to use it here, and specifically chose it over `signal` to bias is slightly towards investigation terminology, but I'm not sure it's the right in the long term. Leaving for now.

File: python-threatexchange/threatexchange/cli/match_cmd.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Dcallies) [reply python-threatexchange/threatexchange/cli/match_cmd.py:72]: I did accidentally a word

**Pull request merged**